backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/BPAAttributeGroup.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.model;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.micronaut.core.annotation.Introspected;
+import lombok.*;
+import org.hyperledger.bpa.controller.api.prooftemplates.AttributeGroup;
+import org.hyperledger.bpa.impl.verification.prooftemplates.DistinctAttributeNames;
+import org.hyperledger.bpa.impl.verification.prooftemplates.ValidAttributeGroup;
+import org.hyperledger.bpa.impl.verification.prooftemplates.ValidBPASchemaId;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Introspected
+@ValidAttributeGroup
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
+public class BPAAttributeGroup {
+    @ValidBPASchemaId
+    String schemaId;
```

Comment:
depends on how the UI looks like, but I think this should not be the schemaId but the reference to `BPASchema.id`

Reply:
@etschelp  can we solve this together?

Reply:
@jaredweinfurtner  will introduce the SchemaAPI here instead.